r1cs/gadgets/std/src/groups/mod.rs

```diff
@@ -239,13 +434,16 @@ pub(crate) mod test {
         let _ = b
             .to_bytes_strict(&mut cs.ns(|| "b ToBytes Strict"))
             .unwrap();
+        assert!(cs.is_satisfied());
+
+        scalar_bits_to_constant_length_test::<<ConstraintF as Field>::BasePrimeField, G>();
     }
 
     #[allow(dead_code)]
-    pub(crate) fn group_test_with_unsafe_add<
+    pub(crate) fn group_test_with_incomplete_add<
```

Comment:
Why do both test, `group_test` and `group_test_with_incomplete_add`, use the incomplete addition?

Reply:
Well actually, one is for curves with complete addition and the other for curves with incomplete additions (some test cases are skipped, e.g. a + 0 = a)

Reply:
Hm. I couldn't tell from the declaration and the code, but maybe that is due to the fact that incomplete addition uses the same call `add()`. In any case let us document that fact, so it is clear to the reader.

Reply:
Yep. We are using GroupGadget trait functions so we don't have access to add_unsafe() from here. A comment was added.